### PR TITLE
Add TO Client API for Origin Automation

### PR DESCRIPTION
**Pull request by dmohan001c**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Control Client 
- Traffic Ops
- CI tests

## What is the best way to verify this PR?
Execute all the Integration tests and make sure the tests passed.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/testing/api/v4/origins_test.go [hunk 1]
```diff
@@ -374,3 +473,411 @@ func DeleteTestOrigins(t *testing.T) {
...
+	//Get Origins by Tenant
+	opts.QueryParameters.Del("profileId")
+	opts.QueryParameters.Set("tenant", "12345")
+	originByTenant, _, _ := TOSession.GetOrigins(opts)
+	if len(originByTenant.Response) > 0 {
+		t.Fatalf("Expected empty response for GET Origin by invalid Tenant, but found %d", len(originByTenant.Response))
+	}
+}
+
+func CreateTestOriginInvalidData(t *testing.T) {
```
Comment (rimashah25) [inline traffic_ops/testing/api/v4/origins_test.go:622]: In this function, shouldn't one test for err first, before checking for status code?, L645, L658, L671, L684, L697, L710, L723

File: traffic_ops/testing/api/v4/origins_test.go [hunk 2]
```diff
@@ -374,3 +473,411 @@ func DeleteTestOrigins(t *testing.T) {
...
+	originByTenant, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Tenant ID: %v - alerts: %+v", err, originByTenant.Alerts)
+	}
+	if len(originByTenant.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Tenant, but found %d", len(originByTenant.Response))
+	}
+}
+
+func GetTestOriginsByInvalidParams(t *testing.T) {
```
Comment (rimashah25) [inline traffic_ops/testing/api/v4/origins_test.go:580]: Do we need to test negative cases for `name`, `primary` and `dsid`?

**Comment by rimashah25** [comment]
Api/v4 and E2E-Integration tests pass on local.
A couple of minor comments.

File: traffic_ops/testing/api/v4/origins_test.go (hunk 1 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/origins_test.go:622]: Do you want me to change the order. Either fail, expected would be the same.

File: traffic_ops/testing/api/v4/origins_test.go (hunk 2 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/origins_test.go:580]: I don't have negative tests for this scenario. I will add.

File: traffic_ops/testing/api/v4/origins_test.go (hunk 1 above)
Comment (rimashah25) [reply traffic_ops/testing/api/v4/origins_test.go:622]: No, I'm ok with order. Normally, I would check for error first before check for any response or reqInf.

File: traffic_ops/testing/api/v4/origins_test.go (hunk 2 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/origins_test.go:580]: Added negative test for the scenarios.

**Review by rimashah25 — APPROVED** [review]
Code LGTM.
